### ISSUE TITLE
`espressif_board_reset_pin_number()` is not working right for ESP32-S2 Feather TFT GPIO21 reset to high

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#6369]
#6272 fixes #6264, but the code seems redundant: we thought `espressif_board_reset_pin_number()` code was supposed to be doing this reset to high already. Diagnose why `espressif_board_reset_pin_number()` is not producing the desired effect.

**Comment by anecdata** [comment]
Is this related? I don't seem to be able to set up the pun to go low-power.

```
Adafruit CircuitPython 7.3.2 on 2022-07-20; Adafruit Feather ESP32-S2 TFT with ESP32S2
>>> import board, digitalio
>>> i2c_tft_power = digitalio.DigitalInOut(board.TFT_I2C_POWER)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: TFT_I2C_POWER in use
```

**Comment by DJDevon3** [comment]
The ESP32-S2 feather guide https://learn.adafruit.com/adafruit-esp32-feather-v2/power-management says deep sleep is activated by turning NEOPIXEL_I2C_POWER pin low. On the ESP32-S2 TFT feather an additional regulator was added and should be triggered by turning off TFT_I2C_POWER which is traced back to pin GPIO21 on the espressif module. 

There is a difference how deep sleep is entered on the feather ESP32-S2 and feather ESP32-S2 TFT.
![ap2112](https://user-images.githubusercontent.com/49322231/181783632-74cb2610-9268-429f-84e8-8abbc4869b83.PNG)

![esp32-s2_TFT](https://user-images.githubusercontent.com/49322231/181783655-12508ed9-866d-4fd7-ade4-47f2c293e6be.png)

Hope that helps.

**Comment by ladyada** [comment]
please note that turning off the regulator *does not trigger deep sleep mode* - only the deep sleep command does that. turning off the regulator *will reduce power greatly which will make entering deep sleep use the minimal power possible* (approx 100uA) 
@kattni plz check guides "The low power section is also missing for the S3 in its learn guide." etc

**Comment by dhalbert** [comment]
There is a lot in this issue which is now addressed by #6907.
